piet-cairo/src/lib.rs

```diff
@@ -319,17 +327,27 @@ fn draw_image<'a>(
     dst_rect: Rect,
     interp: InterpolationMode,
 ) {
+    let src_rect = match src_rect {
+        Some(src_rect) => src_rect,
+        None => Size::new(image.get_width() as f64, image.get_height() as f64).to_rect(),
+    };
+    // Cairo returns an error if we try to paint an empty image, causing us to panic. We check if
+    // either the source or destination is empty, and early-return if so.
+    if src_rect.width() == 0.
```

Comment:
Is there no `Rect::is_empty`? Maybe just `src_rect().area() == 0.0`?

Reply:
Hmm, I could add a `Rect::is_empty`. The `area` solution will work also. I'll have a look at this after I've stopped windows/mac crashing.